### PR TITLE
Switch to mobc pool

### PR DESCRIPTION
**Pull request by pimeys**
The author of mobc is super responsive and the library fits like a glove to Quaint. Now we also get rid of direct tokio dependency and can support async-std.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -67,14 +64,12 @@ pub enum Error {
...
-#[cfg(any(
-    feature = "mysql-16",
-    feature = "postgresql-0_16",
-    feature = "rusqlite-0_19"
-))]
-impl From<r2d2::Error> for Error {
-    fn from(e: r2d2::Error) -> Error {
-        Error::ConnectionError(e.into())
+impl From<mobc::Error<failure::Compat<Error>>> for Error {
+    fn from(e: mobc::Error<failure::Compat<Error>>) -> Self {
```
Comment (tomhoule) [inline src/error.rs]: It's going to get nicer when we switch away from failure. Maybe it's something we should consider doing before 0.1?

File: src/lib.rs [hunk 2]
```diff
@@ -215,29 +215,29 @@ impl Quaint {
...
+            .await?;
+
+        Ok(Self { inner, connection_info })
     }
 
     /// The number of connections in the pool.
-    pub fn capacity(&self) -> usize {
-        self.inner.capacity()
+    pub async fn capacity(&self) -> u32 {
+        self.inner.state().await.connections
```
Comment (tomhoule) [inline src/lib.rs]: it feels a bit weird that you have to await to get the pool's internal state, but it's fine

File: src/error.rs (hunk 1 above)
Comment (pimeys) [reply src/error.rs]: Naturally.

File: src/lib.rs (hunk 2 above)
Comment (pimeys) [reply src/lib.rs]: The state is blocked by an async mutex.

Comment (tomhoule) [reply src/lib.rs]: makes sense

**Pull request merged**